TESTING.rst

```diff
@@ -55,8 +55,33 @@ Follow the guidelines when writing unit tests:
   tests, so we run Pytest with ``--disable-warnings`` but instead we have ``pytest-capture-warnings`` plugin that
   overrides ``recwarn`` fixture behaviour.
 
-**NOTE:** We plan to convert all unit tests to standard "asserts" semi-automatically, but this will be done later
-in Airflow 2.0 development phase. That will include setUp/tearDown/context managers and decorators.
+
+.. note::
+
+  We are in the process of convert all unit tests to standard "asserts" and pytest fixtures
+  so if you find some tests that are still using classic setUp/tearDown approach or unittest asserts, feel
+  free to convert them to pytest.
+
+Airflow configuration for unit tests
+------------------------------------
+
+Some of the unit tests require special configuration set as "default" one. This is done automatically by
+automatically adding ``AIRFLOW__CORE__UNIT_TEST_MODE=True`` to the environment variables in Pytest auto-used
+fixture. This in turn makes airflow configuration load test configuration from
+``airflow/config_templates/config_unit_test.yml`` - defaults from the test configuration replace the original
+defaults stored in ``airflow/config_templates/config.yml``. If you want to add some test-only configuration,
+you should copy the option definition to ``config_unit_test.yml`` and change default value there.
+
+.. note::
+
+  The test configuration for Airflow before July 2023 was automatically generated in a file named
+  ``AIRFLOW_HOME/unittest.cfg`` and template for it was stored in "config_templates" next to the yaml file,
+  however this was only done for the first time you run airflow and you had to manually maintain the file,
+  and it was pretty cumbrsome and arcane knowledge, also this file has been overwritten in Breeze environment
+  with another CI-specific file. Using ````config_unit_test.yml`` as a single source of configuration for
+  tests is much more convenient and it is automatically used by Pytest. If you have the automatically
+  generated ``unittest.cfg`` file it will be automatically removed when you run airflow.
+
```

Comment:
```suggestion
  The test configuration for Airflow before July 2023 was automatically generated in a file named
  ``AIRFLOW_HOME/unittest.cfg`` and the template for it was stored in "config_templates" next to the yaml file. This was only done for the first time you run Airflow and you had to manually maintain the file, which was pretty cumbersome and arcane knowledge. This file has been overwritten in the Breeze environment with another CI-specific file. Using ````config_unit_test.yml`` as a single source of configuration for tests is much more convenient and it is automatically used by Pytest. If you have the automatically  generated ``unittest.cfg`` file it will be automatically removed when you run Airflow.

```